ui/src/dashboards/components/DashboardHeaderEdit.js

```diff
@@ -8,28 +8,46 @@ class DashboardEditHeader extends Component {
     const {dashboard: {name}} = props
     this.state = {name}
     this.handleChange = ::this.handleChange
+    this.handleFormSubmit = ::this.handleFormSubmit
+    this.handleKeyUp = ::this.handleKeyUp
   }
 
   handleChange(name) {
     this.setState({name})
   }
 
+  handleFormSubmit(e) {
+    e.preventDefault()
+    const name = e.target.name.value
+    this.props.onSave(name)
+  }
+
+  handleKeyUp(e) {
+    const {dashboard: {name}, onCancel} = this.props
+    if (e.key === 'Escape') {
+      this.setState({name})
```

Comment:
Oh! I had to think about that for a sec. onCancel actually unmounts the component, which, upon being mounted again, resets its state to what's specified in the constructor. Fair point, @jaredscheib ! This is why PR review is so important. 👍